### PR TITLE
Add support for basic hash/object conversions

### PR DESCRIPTION
**Pull request by superstator**
Addresses #169

### CODE REVIEW

File: examples/hash/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+return json object
```
Comment (simlay) [inline examples/hash/README.md:1]: I'm confused by this README.

**Review by simlay — CHANGES_REQUESTED** [review]
Hi there! Thanks for the contribution!

At first glance, I think this is pretty good. I've got a few nits.

- [ ] Add to the [`CHANGELOG.md`](https://github.com/infinyon/node-bindgen/blob/master/CHANGELOG.md).
- [ ] Remove or make add more of a description to the `README.md` in the hash example.

I don't wanna expand the scope of this task too much but I wonder if we care about `impl<'a, T> JSValue<'a> for  HashMap<U, T> where U: Eq + Hash + JSValue, T: JSValue` (and `TryIntoJS`). Thoughts?

**Comment by superstator** [comment]
Ha, I used the array example as a template and obviously missed some cruft. I will update the readme and changelog, and give the additional impl some thought.

**Comment by superstator** [comment]
I played around a little with the idea of a generic hash key type, but I think it would effectively mean needing to support `Map()` on the JS side, which is not a slam dunk via NAPI. Doable, but maybe best left until somebody has a strong need for it.

**Review by simlay — COMMENTED** [review]
LGTM! Thanks!

**Review by digikata — APPROVED** [review]

**Comment by yannleretaille** [comment]
Was there a reason this wasn't merged? It'd be very useful to have native support for HashMap.

**Pull request closed without merging**